### PR TITLE
Disable hashie property override warnings

### PR DESCRIPTION
**Pull request by adamdecaf**
Hashie recently introduced a change that warns if you construct a
Hashie::Mash with properties which override an existing method. This
can come about with json that contains 'zip' (and other conflicting
names).

There are other gems starting to do the same. https://github.com/chef/chef-zero/pull/276 

https://github.com/intridea/hashie/issues/423
https://github.com/intridea/hashie/blob/master/UPGRADING.md#upgrading-to-352

### CODE REVIEW

**Comment by iMacTia** [comment]
Hi @adamdecaf and thanks for opening this PR.
I'm trying to understand the exact scenario where this warning could pop-up.
My understanding is that it can only happen if:

1. The user is using the mashify middleware (i.e. adding it to the faraday stack).
2. The user is parsing a response containing a key that conflicts with mashify pre-defined methods.

If that's the case, then I'd be inclined to reject the PR as I disagree with this patch being applied to Faraday directly. Very few people will see the warning anyway and some of those who don't see it might end up with having a bug due to the conflict and not being aware of it.
Moreover, I personally think that the maintainer of the hashify gem had a good reason to put the warning in place and we shouldn't take the freedom to silence it just adding `faraday_middleware` to the Gemfile.

So in short, I think it should be up to the users to apply this patch on their application to silence the warning.

Hope this makes sense?

**Comment by adamdecaf** [comment]
Yea. It's probably a change I should make in my code. The problem is I have a ruby cli which is parsing json with `zip` as a field name. This conflicts with what hashify sets. 

I was trying to just do this in my app, but was running into a problem with `mash_class`. I couldn't quite figure out how to override that, but I have another idea I can try.

**Comment by adamdecaf** [comment]
I'm going to close this. Figured it out after taking a break from the screen.

```ruby
class Mash < Hashie::Mash
    disable_warnings
  end

  class Mashify < Faraday::Response::Mashify
    def initialize(app = nil, opts = {})
      super(app, opts.merge({ :mash_class => API::Mash }))
    end
  end
```

**Comment by iMacTia** [comment]
Thank you for reporting back @adamdecaf, I'm sure this will turn useful for others in the future 👍 !
